packages/contracts/src/dollar/core/ERC1155Ubiquity.sol

```diff
@@ -15,50 +15,72 @@ import "../../../src/dollar/utils/SafeAddArray.sol";
 /// - ERC1155 minter, burner and pauser
 /// - TotalSupply per id
 /// - Ubiquity Manager access control
-contract ERC1155UbiquityForDiamond is
-    ERC1155,
-    ERC1155Burnable,
-    ERC1155Pausable
-{
+contract ERC1155Ubiquity is ERC1155, ERC1155Burnable, ERC1155Pausable {
     using SafeAddArray for uint256[];
 
-    IAccessControl public immutable accessCtrl;
+    IAccessControl public accessCtrl;
     // Mapping from account to operator approvals
     mapping(address => uint256[]) private _holderBalances;
     uint256 private _totalSupply;
 
     // ----------- Modifiers -----------
-    modifier onlyMinter() {
+    modifier onlyMinter() virtual {
         require(
             accessCtrl.hasRole(GOVERNANCE_TOKEN_MINTER_ROLE, msg.sender),
             "ERC1155Ubiquity: not minter"
         );
         _;
     }
 
-    modifier onlyBurner() {
+    modifier onlyBurner() virtual {
         require(
             accessCtrl.hasRole(GOVERNANCE_TOKEN_BURNER_ROLE, msg.sender),
             "ERC1155Ubiquity: not burner"
         );
         _;
     }
 
-    modifier onlyPauser() {
+    modifier onlyPauser() virtual {
         require(
             accessCtrl.hasRole(PAUSER_ROLE, msg.sender),
             "ERC1155Ubiquity: not pauser"
         );
         _;
     }
 
+    modifier onlyAdmin() {
+        require(
+            accessCtrl.hasRole(DEFAULT_ADMIN_ROLE, msg.sender),
+            "ERC20Ubiquity: not admin"
+        );
+        _;
+    }
+
     /**
      * @dev constructor
      */
     constructor(address _diamond, string memory uri) ERC1155(uri) {
         accessCtrl = IAccessControl(_diamond);
     }
 
+    /// @notice getDiamond returns the diamond address
+    /// @return diamond address
+    function getDiamond() external view returns (address) {
+        return address(accessCtrl);
+    }
+
+    /// @notice setDiamond update the diamond address
+    /// @param _diamond new diamond address
+    function setDiamond(address _diamond) external onlyAdmin {
+        accessCtrl = IAccessControl(_diamond);
+    }
+
+    /// @notice setURI update the URI
+    /// @param newURI new URI
+    function setURI(string memory newURI) external onlyAdmin {
+        _setURI(newURI);
+    }
```

Comment:
I noticed this was changed from `setUri` elsewhere in the code to now `setURI` 

To maintain consistent code styling, I think we should keep the `setUri` style to match the refactors we did with `NFT` to `Nft`